anndata/_core/anndata.py

```diff
@@ -395,6 +395,9 @@ def _init_as_actual(
         # various ways of initializing the data
         # ----------------------------------------------------------------------
 
+        # If X is a data frame, we store its indices for verification
+        x_indices = []
```

Comment:
Is there a way to write this without having to remember a variable 90 lines later?

Reply:
I think a different way of doing this could be that we convert obs and var first, and then do our checks, but I’m happy enough with this as it is.